### PR TITLE
changed the wrapping function for HTML-bold and italic

### PR DESCRIPTION
**Pull request by keshav234156**
Concerns https://github.com/publiclab/PublicLab.Editor/issues/307

### CODE REVIEW

**Comment by keshav234156** [comment]
Hi have made the new wrapping function for bold, so that other modules are undisturbed It should most probably fix the publiclab/PublicLab.Editor#307 and may also cause some other bugs due to changes. So need to test it with different test cases as much as possible.

Another thing is that to exit out of module we need to press hotkey ie ```Ctrl + B``` for bold and       
 ``` ->``` after that. Though -> may look inconsistent but that because we are using Zero width space, which I believe has its own pros and cons and that was the only solution that I could think of to exit out of code module

**Comment by keshav234156** [comment]
cc: @jywarren @emilyashley @VladimirMikulic @shreyaa-s @NitinBhasneria @Shulammite-Aso

**Comment by keshav234156** [comment]
Mostly this part of code was changed: https://github.com/jywarren/woofmark/pull/1/files#diff-723d030cca36e1fdbae13e84f3ac27b9R20-R40

**Review by VladimirMikulic — APPROVED** [review]
Good, but needs to be documented for future devs.

**Comment by keshav234156** [comment]
@VladimirMikulic Will surely do that, just trying now to get it tested with as much as possible with manual test cases

File: src/html/boldOrItalic.js [hunk 1]
```diff
@@ -4,7 +4,127 @@ var strings = require('../strings');
...
+    chunks.before = chunks.before.replace(rleading, '');
+    chunks.after = chunks.after.replace(rtrailing, '');
+  }
+  else if(!chunks.selection) {
+  	if(!trail && !lead && surrounded(chunks,tag)){
+  		chunks.selection = placeholders;
+        chunks.after = close + '>' + chunks.after;
+        chunks.before += open + '>';
+  	}
+  	else if(!trail && !lead surrounded(chunks, tag)) {
```
Comment (NitinBhasneria) [inline src/html/boldOrItalic.js]: Getting error here./
I think it should be `  	else if(!trail && !lead && surrounded(chunks, tag)) {  `

File: src/html/boldOrItalic.js [hunk 2]
```diff
@@ -4,7 +4,127 @@ var strings = require('../strings');
...
+  		chunks.before = chunks.before + close + '>'
```
Comment (NitinBhasneria) [inline src/html/boldOrItalic.js]: add semicolon.

File: src/html/boldOrItalic.js [hunk 3]
```diff
@@ -4,7 +4,127 @@ var strings = require('../strings');
 var wrapping = require('./wrapping');
```
Comment (NitinBhasneria) [inline src/html/boldOrItalic.js]: wrapping is unused hence can be removed.

**Review by NitinBhasneria — CHANGES_REQUESTED** [review]

**Comment by NitinBhasneria** [comment]
Also, I want to add. 
When we are erasing everything and writing a new line the bold button is nearly not working. Have a look at this.
woofmark1

File: src/html/boldOrItalic.js [hunk 4]
```diff
@@ -67,13 +66,13 @@ function wrapping_bold_italic(tag, placeholders, chunks){
...
         chunks.after = open + '>' + chunks.after;
       }
-    }
-    if (!closebounded(chunks, tag)) {
-      chunks.after = close + '>' + chunks.after;
-      chunks.before += open + '>';
+      }
+      if (!closebounded(chunks, tag)) {
+    	chunks.after = close + '>' + chunks.after;
+      	chunks.before += open + '>';
```
Comment (shreyaa-s-zz) [inline src/html/boldOrItalic.js]: Extra space at the beginning.

File: src/html/boldOrItalic.js [hunk 5]
```diff
@@ -94,8 +93,8 @@ function wrapping_bold_italic(tag, placeholders, chunks){
       chunks.before = chunks.before.replace(/(<[^>]+(?: [^>]*)?>)$/, '</' + tag + '>$1');
     }
   }
-}
 
+}
```
Comment (shreyaa-s-zz) [inline src/html/boldOrItalic.js:93]: The bracket should be shifted to the line above or a line added after closing of bracket and before starting of a new function I think. Rest looks good to me. Great Work!! :smile:

File: src/html/boldOrItalic.js [hunk 6]
```diff
@@ -1,10 +1,129 @@
 'use strict';
 
 var strings = require('../strings');
-var wrapping = require('./wrapping');
 
 function boldOrItalic (chunks, type) {
-  wrapping(type === 'bold' ? 'strong' : 'em', strings.placeholders[type], chunks);
+  wrapping_bold_italic(type === 'bold' ? 'strong' : 'em', strings.placeholders[type], chunks);
```
Comment (Shulammite-Aso) [inline src/html/boldOrItalic.js]: Will it be best to stick to the naming style on the codebase? camelCase in this case.

**Comment by Shulammite-Aso** [comment]
What is `->` please?🙈 Enter?? 

Also,  can you add a GIF of how the changes now works? 

Thanks😆

**Comment by keshav234156** [comment]
> What is `->` please? Enter??
> 
> Also, can you add a GIF of how the changes now works?
> 
> Thanks

```->``` is Right arrow. Will make GIF soon

File: src/html/boldOrItalic.js (hunk 6 above)
Comment (keshav234156) [reply src/html/boldOrItalic.js]: ok

File: src/html/boldOrItalic.js (hunk 4 above)
Comment (keshav234156) [reply src/html/boldOrItalic.js]: Just now implementing the linting tool which should fix this as well

**Comment by keshav234156** [comment]
@Shulammite-Aso @NitinBhasneria @shreyaa-s Please review and try to test it with as much as possible with manual test cases

**Comment by NitinBhasneria** [comment]
@keshav234156 it is a little difficult thing to understand all your changes. As @Shulammite-Aso said can you make the gif of the changes so that we can particularly check on the changes and its effect.
Thank you

**Comment by jywarren** [comment]
Oh interesting! Are you able to publish this on your `gh-pages` branch so we can try it out live? And, can you clarify about needing <kbd>ctrl-B</kbd> AND <kbd>-></kbd>? why not just <kbd>-></kbd>? 

Thank you!

File: src/html/boldOrItalic.js [hunk 7]
```diff
@@ -1,10 +1,126 @@
 'use strict';
 
 var strings = require('../strings');
-var wrapping = require('./wrapping');
 
-function boldOrItalic (chunks, type) {
-  wrapping(type === 'bold' ? 'strong' : 'em', strings.placeholders[type], chunks);
+function boldOrItalic(chunks, type) {
+  wrappingBoldItalic(type === 'bold' ? 'strong' : 'em', strings.placeholders[type], chunks);
```
Comment (jywarren) [inline src/html/boldOrItalic.js:6]: Can you link us to the general `wrapping()` function for comparison? Thanks!

Comment (keshav234156) [reply src/html/boldOrItalic.js:6]: https://github.com/jywarren/woofmark/blob/master/src/html/wrapping.js

**Comment by keshav234156** [comment]
@jywarren I tried deploying gh-page at https://keshav234156.github.io/woofmark-testing/examples/index.html But there is an error.Can you please help me to figure it out?

**Comment by keshav234156** [comment]
@jywarren wrapping function is only called when we press the respective hotkey, for bold its ```Ctrl + B``` and we need to press ```->``` as we have added zero width space ie &#8203. This was the only way I found to exit out of the module.

**Comment by keshav234156** [comment]
Untitled_ Jun 8, 2020 4_43 PM

**Comment by keshav234156** [comment]
@shreyaa-s @NitinBhasneria @Shulammite-Aso  any changes except documenting it.

**Comment by cypherean** [comment]
Since you're familiar with this part of the code, please have a look at this discussion and see if you can answer some of these questions https://github.com/publiclab/PublicLab.Editor/issues/457. Any input would be appreciated! Thanks!

**Comment by jywarren** [comment]
This is looking great. In keeping with our idea of maintaining our own branch, would you be able to open this against `jywarren/woofmark`'s `plots2` branch, so that we have the `master` branch here still tracking the upstream `bevacqua/woofmark`? Does that make sense? I think it does but please correct me if you have a better idea, thank you!!!

**Comment by keshav234156** [comment]
> This is looking great. In keeping with our idea of maintaining our own branch, would you be able to open this against `jywarren/woofmark`'s `plots2` branch, so that we have the `master` branch here still tracking the upstream `bevacqua/woofmark`? Does that make sense? I think it does but please correct me if you have a better idea, thank you!!!

ok,made a new PR #4

**Comment by keshav234156** [comment]
> Since you're familiar with this part of the code, please have a look at this discussion and see if you can answer some of these questions [publiclab/PublicLab.Editor#457](https://github.com/publiclab/PublicLab.Editor/issues/457). Any input would be appreciated! Thanks!

will take a look!!

**Comment by jywarren** [comment]
Deferring to copy of this in #4, moving over there!

**Pull request closed without merging**